src/proposals/20231011/SparkEthereum_20231011.t.sol

```diff
@@ -0,0 +1,208 @@
+        assertEq(emode.liquidationThreshold, 0);
+        assertEq(emode.liquidationBonus,     0);
+        assertEq(emode.priceSource,          address(0));
+        assertEq(emode.label,                '');
+
+        /*******************************/
+        /*** sDAI before validations ***/
+        /*******************************/
+
+        assertEq(_findReserveConfigBySymbol(allConfigsBefore, 'sDAI').eModeCategory, 0);
+
+        /*******************************/
+        /*** rETH before validations ***/
+        /*******************************/
+
+        assertEq(_findReserveConfigBySymbol(allConfigsBefore, 'rETH').supplyCap, 20_000);
+
+        /**************************************/
+        /*** USDC & USDT before validations ***/
+        /**************************************/
+
+        ReserveConfig memory usdcConfigBefore = _findReserveConfigBySymbol(allConfigsBefore, 'USDC');
+        assertTrue(usdcConfigBefore.aToken            != address(0));
+        assertTrue(usdcConfigBefore.variableDebtToken != address(0));
+        assertTrue(usdcConfigBefore.stableDebtToken   != address(0));
+
+        // There should be 8 markets before USDT
+        assertEq(allConfigsBefore.length, 8);
+
+        /***********************/
+        /*** Execute Payload ***/
+        /***********************/
+
+        GovHelpers.executePayload(vm, payload, executor);
+
+        ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot('', pool);
+
+        /***********************************/
+        /*** USD eMode after validations ***/
+        /***********************************/
+
+        emode = pool.getEModeCategoryData(2);
+        assertEq(emode.ltv,                  91_00);
+        assertEq(emode.liquidationThreshold, 92_00);
+        assertEq(emode.liquidationBonus,     101_00);
+        assertEq(emode.priceSource,          address(0));
+        assertEq(emode.label,                'USD');
+
+        /******************************/
+        /*** sDAI after validations ***/
+        /******************************/
+
+        assertEq(_findReserveConfigBySymbol(allConfigsAfter, 'sDAI').eModeCategory, 2);
+
+        /******************************/
+        /*** rETH after validations ***/
+        /******************************/
+
+        assertEq(_findReserveConfigBySymbol(allConfigsAfter, 'rETH').supplyCap, 60_000);
+
+        /*************************************/
+        /*** USDC & USDT after validations ***/
+        /*************************************/
+
+        // There should be 9 markets after adding USDT
+        assertEq(allConfigsAfter.length, 9);
+
+        // USDT
+        ReserveConfig memory usdtConfigAfter = _findReserveConfigBySymbol(allConfigsAfter, 'USDT');
+        ReserveConfig memory usdtConfigExpected = ReserveConfig({
+            symbol:                  'USDT',
+            underlying:               SparkEthereum_20231011(payload).USDT(),
```

Comment:
Can we set all these calls to the payload assertions to manual values? Will be safter

Reply:
I debated this myself, but not sure how much this actually helps. I suppose if there is a mistake in the spell that all of us happen to miss it reduces the chances of mistake double checking in another place.